libtenzir/builtins/operators/compress_decompress.cpp

```diff
@@ -388,10 +450,25 @@ class compress_plugin final : public virtual operator_plugin<compress_operator>,
     -> failure_or<operator_ptr> override {
     auto args = operator_args{};
     auto level = std::optional<located<int64_t>>{};
-    TRY(argument_parser2::operator_(name())
-          .positional("type", args.type)
-          .named("level", level)
-          .parse(inv, ctx));
+    auto parser = argument_parser2::operator_(name());
+    if (method_name_.empty()) {
+      parser.positional("type", args.type);
+    }
+    parser.named("level", level);
+    if (method_name_ == "gzip") {
+      parser.named("format", args.gzip_format);
+    }
+    auto window_bits = std::optional<located<uint64_t>>{};
+    if (method_name_ == "gzip" or method_name_ == "brotli") {
+      parser.named("window_bits", window_bits);
+    }
+    TRY(parser.parse(inv, ctx));
+    if (method_name_.empty()) {
+      diagnostic::warning(R"(`{} "{}"` is deprecated)", name(), args.type.inner)
+        .hint("Use `{}_{}` instead", name(), args.type.inner)
```

Comment:
```suggestion
        .hint("use `{}_{}` instead", name(), args.type.inner)
```

Don't we use lowercase for diagnostics?